bleak/backends/bluezdbus/scanner.py

```diff
@@ -77,8 +78,31 @@ def __init__(self, **kwargs):
     async def start(self):
         self._bus = await MessageBus(bus_type=BusType.SYSTEM).connect()
 
+        # Find the HCI device to use for scanning and get cached device properties
+        reply = await self._bus.call(
```

Comment:
Moving this before attaching the signal listeners will cause a race condition where signals can be missed, so unfortunately this is not going to work.